File: sesi15/konversi_hari.js
console.log('program mengkonversi proyek hari ke Tahun dan Bulan')

const readline = require('node:readline').createInterface({
  input: process.stdin,
  output: process.stdout
})

const satuTahunDalamHari = 365
const satuBulanDalamHari = 30

readline.question('masukkan jumlah hari proyek: ', jumlah_hari_proyek => {
  let tahun = Math.floor(jumlah_hari_proyek / satuTahunDalamHari) // dapat nilai tahun
  let sisaHariSetelahTahun = jumlah_hari_proyek % satuTahunDalamHari // sisa dari tahun
  let bulan = Math.floor(sisaHariSetelahTahun / satuBulanDalamHari) // dapat nilai bulan
  let sisaHariSetelahBulan = sisaHariSetelahTahun % satuBulanDalamHari // dapat sisa hari
  let hari = sisaHariSetelahBulan
  console.log(`Hasil dari konversi jumlah hari proyek ${jumlah_hari_proyek} yaitu: ${tahun} tahun, ${bulan} bulan, dan ${hari} hari`)
  readline.close()
})

File: sesi15/konversi_hari.rb
puts "program mengkonversi hari ke tahun dan bulan"
print "masukkan proyek dalam hari: "
proyek_dalam_hari = gets.chomp.to_i
# divmod
tahun_dalam_hari = 365
bulan_dalam_hari = 30
tahun_dan_sisa_hari = proyek_dalam_hari.divmod(tahun_dalam_hari) # [1, 35]
tahun = tahun_dan_sisa_hari.first
bulan_dan_sisa_hari = tahun_dan_sisa_hari.last.divmod(bulan_dalam_hari) # [1, 5]
bulan = bulan_dan_sisa_hari.first
hari = bulan_dan_sisa_hari.last
puts "Hasil konversi dari hari proyek #{proyek_dalam_hari} yaitu: #{tahun} tahun, #{bulan} bulan, dan #{hari} hari."
